fix: revert uniform implementation and remove unnecessary loop

abhishekblue opened the pull request:
---
type: pre_push_report
description: Results of running various checks prior to pushing changes.  
report:
  - task: run_javascript_examples status: na
  - task: run_c_examples status: na
  - task: run_cpp_examples status: na
  - task: run_javascript_readme_examples status: na
  - task: run_c_benchmarks status: na
  - task: run_cpp_benchmarks status: na
  - task: run_fortran_benchmarks status: na
  - task: run_javascript_benchmarks status: na
  - task: run_julia_benchmarks status: na
  - task: run_python_benchmarks status: na
  - task: run_r_benchmarks status: na
  - task: run_javascript_tests status: na
---

# Resolves #5228  


## Description

> What is the purpose of this pull request?

This pull request:

- Reverts `@stdlib/random/base/uniform` back to `@stdlib/random/array/uniform` as per review feedback in fb90f93#r152597788.
- Updates `benchmark.native.js` to match the expected implementation.
- Removes unnecessary for loop and directly initializes the uniform array.

## Related Issues

> Does this pull request have any related issues?

This pull request:

- Address commit comments (commit fb90f93).

## Questions

> Any questions for reviewers?

No.

## Other

> Any other relevant information?

- `make lint` produced filename errors in unrelated files. These files were not modified as part of this PR.
- Skipping lint fixes since they are unrelated to this issue.

## Checklist

- [x] Read, understood, and followed the [contributing guidelines]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

* * *

@stdlib-js/reviewers

abhishekblue committed:
fix: remove unnecessary for loop and update uniform usage

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.native.js:
```diff
@@ -46,10 +46,9 @@ bench( pkg+'::native', opts, function benchmark( b ) {
-		lambda[ i ] = uniform( 1.0, 10.0 );
-	}
+	
+	lambda = uniform(len, 0.1, 10.0);
```
```suggestion
	lambda = uniform( 100, 0.1, 10.0 );
```
as per the suggested changes the `var len` should be removed as the original implementation worked just fine.
Could you pls do the same for the `benchmark.js` file as well.

anandkaranubc replied:
Just adding on to it, once the `len` variable is removed, the initialization of `y` in the benchmark loop will be updated to:  

```js
y = median( lambda[ i % lambda.length ] );
```

Basically, everything that was done in the original PR, as Neeraj already pointed out.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.js:
```diff
@@ -21,7 +21,7 @@
 var bench = require( '@stdlib/bench' );
-var Float64Array = require( '@stdlib/array/float64' );
+var uniform = require( '@stdlib/random/array/uniform' );
 var uniform = require( '@stdlib/random/base/uniform' );
```
```suggestion

```
this should be removed as per the original implementation you will only be needing `@stdlib/random/array/uniform`.

Neerajpathak07 replied:
@anandkaranubc what's your take on this?

anandkaranubc replied:
I agree. The latter one needs to be removed since it is no longer in use. Thanks for pointing that out!

saurabhraghuvanshii replied:
there is lint error in line no 39 , 42

Neerajpathak07 replied:
Yes, I believe clicking on the details should provide an overview for it.

abhishekblue commented:
Hi @Neerajpathak07 , my PR is ready, and I have made all the requested changes. Could you please approve the workflows? Thanks!

Neerajpathak07 commented:
@abhishekblue have you committed and pushed those changes cause they don't reflect here??
And the workflow approvals lie with the maintainers.

abhishekblue committed:
made suggested changes and removed for loop

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

abhishekblue commented:
Hi @Neerajpathak07,
Apologies for the confusion earlier! I had committed the changes but mistakenly thought they were already pushed.
I have now pushed the latest changes. Could you please check if they reflect correctly and approve the workflows if everything looks good? 
I also have to remove for loop from benchmark.js as well. 
Thanks for your patience!

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.js:
```diff
@@ -38,13 +37,12 @@ bench( pkg, function benchmark( b ) {
 
 	len =100;
 	lambda = new Float64Array( len );
```
```suggestion

```
won't be needing the `Float64Array` module so better to remove it

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.js:
```diff
@@ -38,13 +37,12 @@ bench( pkg, function benchmark( b ) {
-		lambda[ i ] = uniform( 1.0, 10.0);
-	}
+
+	lambda = uniform(100, 1.0, 10.0); 
```
```suggestion
	lambda = uniform( 100, 0.1, 10.0 ); 
```

Neerajpathak07 replied:
Need the same change for the `native` version as well.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.native.js:
```diff
@@ -46,14 +46,13 @@ bench( pkg+'::native', opts, function benchmark( b ) {
 	var i;
 
 	len = 100;
```
here as well!

Neerajpathak07 replied:
You'll also want to remove 
```javascript
var uniform = require( '@stdlib/random/base/uniform' );
```
and add
```javascript
var uniform = require( '@stdlib/random/array/uniform' );
```

Neerajpathak07 commented:
@abhishekblue Just a few changes rest everything looks in line to me. Thank you for your efforts to correct them!!

abhishekblue committed:
apply suggested formmating changes and removed unused vairables

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

abhishekblue commented:
@Neerajpathak07  I've made the suggested formatting changes and also removed unused variables. Let me know if anything else is required. 
Thanks!

abhishekblue committed:
update uniform import

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.js:
```diff
@@ -32,19 +31,15 @@ var median = require( './../lib' );
-	for ( i = 0; i < len; i++ ) {
-		lambda[ i ] = uniform( 1.0, 10.0);
-	}
+
```
```suggestion
```
please remove this extra line.
Same for the native file.

abhishekblue committed:
removed extra lines

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented:
LGTM! @kgryte Kindly take a look and merge it eventually!!

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/planck/median/benchmark/benchmark.js:
```diff
@@ -32,19 +31,14 @@ var median = require( './../lib' );
-	for ( i = 0; i < len; i++ ) {
-		lambda[ i ] = uniform( 1.0, 10.0);
-	}
+	lambda = uniform( 100, 1.0, 10.0 ); 
```
```suggestion
	lambda = uniform( 100, 0.1, 10.0 );
```

Something that doesn’t make a huge difference but would be better to have the same range as the original for consistency. :)

Neerajpathak07 replied:
absolutely! had suggested something similar in this [suggestion](https://github.com/stdlib-js/stdlib/pull/5314#discussion_r1963528446) must have been overlooked.

abhishekblue committed:
Updated benchmark.js

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented:
LGTM!!
@kgryte Kindly take a look at this and merge it eventually.

Planeshifter commented:
Thanks @abhishekblue for your PR and to @saurabhraghuvanshii, @Neerajpathak07 and @anandkaranubc for review comments!

Planeshifter committed:
chore: remove trailing space
